xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -584,15 +584,16 @@ float CGUIFontTTF::GetTextWidthInternal(const vecText& text, const std::vector<G
   float width = 0;
   for (auto it = glyphs.begin(); it != glyphs.end(); it++)
   {
-    Character* c = GetCharacter(text[(*it).m_glyphInfo.cluster], (*it).m_glyphInfo.codepoint);
+    const character_t ch = text[(*it).m_glyphInfo.cluster];
+    Character* c = GetCharacter(ch, (*it).m_glyphInfo.codepoint);
     if (c)
     {
       // If last character in line, we want to add render width
       // and not advance distance - this makes sure that italic text isn't
       // choped on the end (as render width is larger than advance then).
       if (std::next(it) == glyphs.end())
         width += std::max(c->m_right - c->m_left + c->m_offsetX, c->m_advance);
-      else if ((c->m_letter & 0xffff) == static_cast<character_t>('\t'))
+      else if ((ch & 0xffff) == static_cast<character_t>('\t'))
```

Comment:
Doing currently a rebase of my font request (https://github.com/xbmc/xbmc/pull/20552).

About here it looks a bit strange, with before was the `& 0xffff` done within a `wchar_t` where can have the color and style bits and makes sense.
But now becomes it done within an structure where `letter` works too, as is the first value inside. But his compare then make always a complete compare, what unneeded works.
Was a reason to use not as?:
```cpp
else if (ch.letter == static_cast<decltype(ch.letter)>('\t'))
```

Also as question about line 587 with `const character_t ch = text[(*it).m_glyphInfo.cluster];`, was a reason to use not as `const character_t& ch` to prevent struct copy?